src/databricks/labs/ucx/azure/credentials.py

```diff
@@ -246,16 +251,27 @@ def run(self, prompts: Prompts, include_names: set[str] | None = None) -> list[S
         plan_confirmed = prompts.confirm(
             "Above Azure Service Principals will be migrated to UC storage credentials, please review and confirm."
         )
+        sp_results = []
         if plan_confirmed:
-            sp_results = self._migrate_service_principals(include_names)
-        else:
-            sp_results = []
+            sp_migration_infos = self._generate_migration_list(include_names)
+            plan_confirmed = True
+            if any(spn.permission_mapping.default_network_action != "Allow" for spn in sp_migration_infos):
+                plan_confirmed = prompts.confirm(
+                    "At least one Azure Service Principal accesses a storage account with non-Allow default network "
+                    "configuration, which might cause connectivity issues. We recommend using Databricks Access "
+                    "Connectors instead (next prompt). Would you like to continue with migrating the service "
+                    "principals?"
+                )
+            if plan_confirmed:
+                sp_results = self._migrate_service_principals(sp_migration_infos)
 
-        plan_confirmed = prompts.confirm("Please confirm to create an access connector for each storage account.")
+        plan_confirmed = prompts.confirm(
+            "[RECOMMENDED] Please confirm to create an access connector with a managed identity for each storage "
```

Comment:
nit on this one - our default quota for storage credential is 200 versus 500 for external location, so creating one access connector per storage account might hit quota limit for customers - we could do address this in a follow up PR though